### DISCUSSION TITLE
ExpressionScope 传递 observable 数据，当数据发生变化时，表达式访问不到最新值

### DISCUSSION THREAD

**Discussion by chenWebCoder** [alibaba/formily#2919] [Q&A 2.x] [answered]
https://codesandbox.io/s/purple-glitter-fr9rdx?file=/App.tsx

如示例，在 ExpressionScope 中设置 $enum、$enum_obs；当修改 obs.$enum_obs 值后，在表达式中无法获取到最新的值！

**Comment by janryWang** [comment, marked as answer]
你设置的$enum_obs是给一个普通对象，然后每次更新又是覆盖式更新，这种怎么可能响应

**Reply by chenWebCoder** [reply]
嗯...... 是需要将 ExpressionScope 的 value 值也变成一个 observable 对象吗？似乎这样也不行！

不是太清楚如何将表达式与一个外部响应式的数据进行关联；方便给个示例吗？🙏

**Reply by janryWang** [reply]
类似于useRef一样，再包一层对象，那一层对象必须是obs，然后对obs的属性做赋值，这样才能触发依赖收集

**Reply by chenWebCoder** [reply]
昨天尝试对传入 ExpressionScope 的 value 值进行 observevable 处理，但是似乎还是没有触发依赖收集！请问是我的打开方式不对吗？

![image](https://user-images.githubusercontent.com/23742222/157794634-760b94ce-db3b-487a-8cd0-eb43fd5a921d.png)

https://codesandbox.io/s/purple-glitter-fr9rdx?file=/App.tsx

**Reply by janryWang** [reply]
useMemo(()=>observable(...),[])

**Reply by chenWebCoder** [reply]
useMemo 不能解决这个问题！总感觉表达式访问 obs 变量时是访问的闭包内缓存的内容；

可以帮忙修改下示例来达到 obs 与 表达式进行关联吗？非常感谢🙏

https://codesandbox.io/s/purple-glitter-fr9rdx?file=/App.tsx

**Reply by janryWang** [reply]
https://codesandbox.io/s/inspiring-haze-p0byfn

**Reply by chenWebCoder** [reply]
哦，反应过来了！！！

ExpressionScope 中直接将 value={obs_container} 看似是将 observable 处理之后的响应式数据赋值给 value，其实本质还是将一个普通的对象赋值给了 value，与最开始的写法并无二异，并没有成功的将 value 值 observable 化！

所以 ExpressionScope 中应该 value={{ obs_container }} 

十分感谢🙏
